src/AuthorizedSocketConnection.js

```diff
@@ -28,6 +28,7 @@ export type AuthorizedSocketOptions<TContext, TCredentials> = {|
   hasPermission: (data: any, credentials: TCredentials) => boolean,
   maxSubscriptionsPerConnection?: number,
   createLogger: CreateLogger,
+  documentValidationRules?: $ReadOnlyArray<any>,
```

Comment:
this is just `validationRules` on express-graphql, right?

Reply:
doesn't `SubscriptionServer` have to explicitly pass this down? also, we need something factory-like... proper validation (which we're not doing) requires passing in the query variables too... see e.g. https://github.com/pa-bru/graphql-cost-analysis (maybe we should just switch to this?... but i don't like the API as much)

see also https://developer.github.com/v4/guides/resource-limitations/

Reply:
yes to being passed down from Server,

I'd stick with matching the graphql-express api for now? I don't think they allow the factory thing, but probably should i guess.

Reply:
graphql-express actually lets you build a new "options" per request, which is how they accomplish this. i don't know what the analogous thing would be here, since a number of these things are already factories